subprojects/build-init/src/main/java/org/gradle/buildinit/plugins/internal/BuildScriptBuilder.java

```diff
@@ -513,11 +529,6 @@ public ChainedPropertyExpression(ExpressionValue left, ExpressionValue right) {
             this.right = right;
         }
 
-        @Override
-        public boolean isBooleanType() {
```

Comment:
💭 All the various small refactors like this default method and extracting methods for common functionality here really improve the code, but they make the core of this change harder to find when reviewing the PR.  Probably better to extract them to a separate PR next time, or if not possible, add some explanatory review comments to point people towards the more important stuff.

Reply:
moved to
- #23374